lib/metadata/index.js

```diff
@@ -125,6 +125,11 @@ Metadata.prototype.buildTokens = function (hosts) {
   this.primaryReplicas = primaryReplicas;
   //All the tokens in ring order
   this.ring = allSorted;
+  //Compute string versions as it's potentially expensive and frequently reused later
+  this.ringTokensAsStrings = [];
+  for (var i = 0; i < allSorted; i++) {
```

Comment:
As you've seen, I added a couple of tests around this. Only covering the ring/ringTokensAsStrings right now as I wouldn't be confident saying what the other values *should* be.